core/controllers/feedback_test.py

```diff
@@ -554,3 +556,216 @@ def test_feedback_threads_with_suggestions(self):
         }
         self.assertDictContainsSubset(
             expected_suggestion_dict, response['suggestion'])
+
+    def test_get_feedback_threads_with_no_text_and_no_updated_status_raises_400(
+            self):
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_html_response('/create/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        feedback_services.create_thread(
+            feconf.ENTITY_TYPE_EXPLORATION, self.EXP_ID, self.owner_id_1,
+            'a subject', 'some text', has_suggestion=True)
+
+        response_dict = self.get_json(
+            '%s/%s' % (feconf.FEEDBACK_THREADLIST_URL_PREFIX, self.EXP_ID))
+        threadlist = response_dict['suggestion_thread_dicts']
+        thread_id = threadlist[0]['thread_id']
+
+        new_content = state_domain.SubtitledHtml(
+            'content', 'new content html').to_dict()
+        change = {
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+            'state_name': 'Welcome!',
+            'new_value': new_content
+        }
+        suggestion_models.GeneralSuggestionModel.create(
```

Comment:
You could create suggestions at the service layer itself, you don't need to create models.

Reply:
Actually I need the thread for the post request and that's not possible with suggestion_services.create_suggestion(since it creates a thread id of its own). This is why I created the model

Reply:
I think you can just create instances of GeneralfeedbackModel (possibly using functions in feedback_services)

Reply:
Hmm if you look at ThreadHandler, you can see we are getting the suggestion as get_suggestion_by_id and get_suggestion_by_id fetches the model from GeneralSuggestionModel, so I don't understand what can I do with GeneralfeedbackModel(btw I think you mean GeneralFeedbackThreadUserModel)

Reply:
Done!